cyclic-position-benchmark/runcycletestsandplot.py

```diff
@@ -0,0 +1,582 @@
+import collections
+import json
+import math
+import os
+import subprocess
+import time
+from threading import Thread
+import sgfmill
+import sgfmill.sgf
+import sgfmill.boards
+import sgfmill.ascii_boards
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+
+PLOTS_PATH = "generated_plots"  # path to output plots
+SGFS_PATH = "sgfs"  # path to input positions
+KATAGO_PATH = "/engines/KataGo-raw/cpp/katago"  # path to compiled KataGo
+MODELS_PATH = "/go_attack/victim_models/"  # path to models to be tested
+
+
+def get_model_name_for_plotting(input_model_path):
+    """
+    Adjust this to get the part of the model name that you want to appear on the plots.
+    E.g., remove extension, remove prefixes all the models have, etc.
+    """
+    return input_model_path.split("/")[-1].split(".bin")[0]
```

Comment:
`pathlib.Path(input_model_path).stem` would accomplish the same thing

Reply:
I could be wrong but I think that would still need some `.split` part when it's e.g. "xx.bin.gz" and not just "xx.bin" - so to me it's clearer what it's doing like this (at least as default, then one can change to fit use case). Will leave it for now but if I'm mistaken or you find the stem way more helpful in practice feel free to change later.